include/z64play.h

```diff
@@ -0,0 +1,555 @@
+#include "z64scene.h"
+#include "z64transition.h"
+#include "z64view.h"
+#include "prerender.h"
+#include "color.h"
+
+typedef struct {
+    /* 0x00 */ u8   seqId;
+    /* 0x01 */ u8   natureAmbienceId;
+} SequenceContext; // size = 0x2
+
+typedef struct {
+    /* 0x00 */ s32 enabled;
+    /* 0x04 */ s32 timer;
+} FrameAdvanceContext; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ Vec3f    pos;
+    /* 0x0C */ f32      unk_0C; // radius?
+    /* 0x10 */ Color_RGB8 color;
+} TargetContextEntry; // size = 0x14
+
+typedef struct {
+    /* 0x00 */ Vec3f    naviRefPos; // possibly wrong
+    /* 0x0C */ Vec3f    targetCenterPos;
+    /* 0x18 */ Color_RGBAf naviInner;
+    /* 0x28 */ Color_RGBAf naviOuter;
+    /* 0x38 */ Actor*   arrowPointedActor;
+    /* 0x3C */ Actor*   targetedActor;
+    /* 0x40 */ f32      unk_40;
+    /* 0x44 */ f32      unk_44;
+    /* 0x48 */ s16      unk_48;
+    /* 0x4A */ u8       activeCategory;
+    /* 0x4B */ u8       unk_4B;
+    /* 0x4C */ s8       unk_4C;
+    /* 0x4D */ char     unk_4D[0x03];
+    /* 0x50 */ TargetContextEntry arr_50[3];
+    /* 0x8C */ Actor*   unk_8C;
+    /* 0x90 */ Actor*   bgmEnemy; // The nearest enemy to player with the right flags that will trigger NA_BGM_ENEMY
+    /* 0x94 */ Actor*   unk_94;
+} TargetContext; // size = 0x98
+
+typedef struct {
+    /* 0x00 */ void*      texture;
+    /* 0x04 */ s16      x;
+    /* 0x06 */ s16      y;
+    /* 0x08 */ u8       width;
+    /* 0x09 */ u8       height;
+    /* 0x0A */ u8       durationTimer; // how long the title card appears for before fading
+    /* 0x0B */ u8       delayTimer; // how long the title card waits to appear
+    /* 0x0C */ s16      alpha;
+    /* 0x0E */ s16      intensity;
+} TitleCardContext; // size = 0x10
+
+typedef struct {
+    /* 0x00 */ s32    length; // number of actors loaded of this category
+    /* 0x04 */ Actor* head; // pointer to head of the linked list of this category (most recent actor added)
+} ActorListEntry; // size = 0x08
+
+typedef struct {
+    /* 0x0000 */ u8     freezeFlashTimer;
+    /* 0x0001 */ char   unk_01[0x01];
+    /* 0x0002 */ u8     unk_02;
+    /* 0x0003 */ u8     lensActive;
+    /* 0x0004 */ char   unk_04[0x04];
+    /* 0x0008 */ u8     total; // total number of actors loaded
+    /* 0x000C */ ActorListEntry actorLists[ACTORCAT_MAX];
+    /* 0x006C */ TargetContext targetCtx;
+    struct {
+        /* 0x0104 */ u32    swch;
+        /* 0x0108 */ u32    tempSwch;
+        /* 0x010C */ u32    unk0;
+        /* 0x0110 */ u32    unk1;
+        /* 0x0114 */ u32    chest;
+        /* 0x0118 */ u32    clear;
+        /* 0x011C */ u32    tempClear;
+        /* 0x0120 */ u32    collect;
+        /* 0x0124 */ u32    tempCollect;
+    }                   flags;
+    /* 0x0128 */ TitleCardContext titleCtx;
+    /* 0x0138 */ char   unk_138[0x04];
+    /* 0x013C */ void*  absoluteSpace; // Space used to allocate actor overlays with alloc type ACTOROVL_ALLOC_ABSOLUTE
+} ActorContext; // size = 0x140
```

Comment:
z64actor.h?

Reply:
I think I'll avoid touching z64actor.h for now, it's kind of a mess. probably should be split eventually